src/compiler/transformers/module/system.ts

```diff
@@ -225,7 +223,7 @@ namespace ts {
 
             // Add any prologue directives.
             const ensureUseStrict = getStrictOptionValue(compilerOptions, "alwaysStrict") || (!compilerOptions.noImplicitUseStrict && isExternalModule(currentSourceFile));
-            const statementOffset = factory.copyPrologue(node.statements, statements, ensureUseStrict, sourceElementVisitor);
+            const statementOffset = factory.copyPrologue(node.statements, statements, ensureUseStrict, topLevelVisitor);

```

Comment:
Is it possible to share code between module.ts and system.ts? They parts in this review look identical.

Reply:
Most of the sharing we do between these two module systems is in `collectExternalModuleInfo`. The problem with refactoring the similarities between the two is that in both cases they close over specific visitors for their individual cases. We could conceivably refactor module emit, but that's not a task I'd like to take on in this PR.